ecommerce/core/management/commands/migrate_program_offers_to_commercetools.py

```diff
@@ -217,57 +331,74 @@ def _group_other_offers(cart_discounts: list):
         })
 
 
+def _get_non_ten_percentage_offer_uuids():
+    """
+    Get non-10% discount offer uuids from legacy ecommerce.
+
+    Args:
+        cart_discounts (set): List of cart discounts.
+    """
+    non_ten_percentage_offer_uuids = [
+        str(uuid) for uuid in ConditionalOffer.objects.filter(
+            offer_type=ConditionalOffer.SITE,
+            condition__program_uuid__isnull=False,
+        ).exclude(
+            benefit__value=10,
+            benefit__proxy_class=ProxyClassDiscountType.PERCENTAGE.value
+        ).values_list('condition__program_uuid', flat=True)
+    ]
+    return set(non_ten_percentage_offer_uuids)
+
+
 def _migrate_program_offers(client):  # pylint: disable=too-many-statements
     """
     Migrate program offers to Commercetools.
 
     Args:
         client (CommercetoolsAPIClient): Commercetools API client.
     """
+    created_discounts = []
+    updated_discounts = []
+    failed_discounts = []
+    deleted_discounts = []
+
     sort_order = _get_highest_sort_order(client)
+    existing_cart_discounts_in_ct = _get_ct_bundle_offers_without_code(client, failed_discounts)
+    non_ten_percentage_offer_uuids = _get_non_ten_percentage_offer_uuids()
 
     cart_discounts = []
     _group_ten_percentage_offers(cart_discounts)
     _group_other_offers(cart_discounts)
 
-    created_discounts = []
-    updated_discounts = []
-    failed_discounts = []
+    # Delete cart discounts that are no longer in legacy ecommerce
+    _delete_extra_ct_bundle_offers(
+        client,
+        cart_discounts,
+        existing_cart_discounts_in_ct,
+        deleted_discounts,
+        failed_discounts
```

Comment:
ideally we should not pass lists from here and mutate them. We should use return value of this function